cmd/bnsd/x/username/init.go

```diff
@@ -0,0 +1,40 @@
+package username
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+// Initializer fulfils the Initializer interface to load data from the genesis
+// file
+type Initializer struct{}
```

Comment:
pp: how about introducing a `type InitializerFunc func(opts weave.Options, params weave.GenesisParams, kv weave.KVStore) error` in weave package  that implements the Initializer interface so that don't have to create empty structs to be compatible?

Reply:
I don't mind this one line that would otherwise be unnecessary. 
As a counter argument, currently there is only one way of providing an initializer and every extension implements it the same way. I think this is simpler to understand.